core/io/image.cpp

```diff
@@ -2051,6 +2051,24 @@ Vector<uint8_t> Image::get_data() const {
 	return data;
 }
 
+Ref<Image> Image::create_empty(int p_width, int p_height, bool p_use_mipmaps, Format p_format) {
+	Ref<Image> image;
+	image.instantiate();
+	image->create(p_width, p_height, p_use_mipmaps, p_format);
+	return image;
+}
+
+Ref<Image> Image::create_from_data(int p_width, int p_height, bool p_use_mipmaps, Format p_format, const Vector<uint8_t> &p_data) {
+	Ref<Image> image;
+	image.instantiate();
+	image->create(p_width, p_height, p_use_mipmaps, p_format, p_data);
+	return image;
+}
+
+void Image::set_data(int p_width, int p_height, bool p_use_mipmaps, Format p_format, const Vector<uint8_t> &p_data) {
+	create(p_width, p_height, p_use_mipmaps, p_format, p_data);
+}
+
 void Image::create(int p_width, int p_height, bool p_use_mipmaps, Format p_format) {
```

Comment:
I think this warrants further refactoring, it's weird to have static `create_empty` and `create_from_data` and non-static `create` with two variants.

Maybe:
- Move `create(int p_width, int p_height, bool p_use_mipmaps, Format p_format)` logic to `create_empty` (and rename to `create` to match bindings)
- Move `create(int p_width, int p_height, bool p_use_mipmaps, Format p_format, const Vector<uint8_t> &p_data)` logic to `set_data` and use it in `create_from_data`.